src/index.js

```diff
@@ -26,7 +26,11 @@ export default class ScrollBehavior {
     /* istanbul ignore else: Travis browsers all support this */
     if ('scrollRestoration' in window.history) {
       this._oldScrollRestoration = window.history.scrollRestoration;
-      window.history.scrollRestoration = 'manual';
+      try {
+        window.history.scrollRestoration = 'manual';
+      } catch () {
+        this._oldScrollRestoration = null;
```

Comment:
You may as well keep around this value, and add a try-catch around the attempt to reset it at the end.